scripts/autoclose.js

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env node
+// @ts-check
+/**
+ * @file Auto-close script for closing won't add icons.
+ */
+import path from 'node:path';
+import process from 'node:process';
+
+/**
+ * @typedef {object} Rule
+ * @property {RegExp[]} patterns - The pattern to match against the issue title.
+ * @property {string} reason - The issue numbers to include in the reason text.
+ */
+
+/**
+ * @typedef {object} Issue
+ * @property {{name: string}[]} labels - Isesue labels.
+ * @property {string} state - Issue state, possible values are 'open' and 'closed'.
+ * @property {string} title - Issue title.
+ * @property {string} body - Issue body.
+ */
+
+/** @type {Rule[]} */
+const rules = await import(
+	path.join(import.meta.dirname, '..', 'autoclose.rules.mjs')
+).then((module) => module.default);
+
+const {GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER} = process.env;
+
+if ([GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER].some((v) => !v)) {
+	console.error(
+		`${Object.entries({GITHUB_TOKEN, GITHUB_REPOSITORY, ISSUE_NUMBER})
+			.filter(([, v]) => !v)
+			.map(([k]) => k)
+			.join(', ')} environment variable(s) must be set.`,
+	);
+	process.exit(1);
+}
+
+/**
+ * Fetch data from GitHub API.
+ * @param {string} url The URL to fetch.
+ * @param {globalThis.RequestInit} options The options to pass to the fetch function.
+ * @returns {Promise<any>} - The response data.
+ */
+const githubFetch = async (url, options) => {
+	const response = await globalThis.fetch(new URL(url), {
```

Comment:
```suggestion
	const response = await globalThis.fetch(url, {
```
Is there a reason we need to create a new `URL` instance from the raw string url? `fetch` accepts a plain string just fine afaik.

Reply:
URL is also available. The benefit is that it can validate the URL string before use.

Reply:
+1 for validating the URL where possible. The amount of times assuming a string will be a URL (in my own stuff, not here) has tripped me up is unbelievable.